# Medical Insurance Cost Dataset Analysis 

Investigation of the Medical Insurance dataset from Kaggle using python functions and classes.

In [1]:
import csv

create empty lists for different columns of the dataset

In [2]:
age = []
sex= []
bmi=[]
children = []
smoker=[]
region =[]
charges= []

read csv file and convert to multiple lists

In [3]:
with open('insurance.csv') as file:
    output = csv.DictReader(file)
    for items in output:
        age.append(items['age'])
        sex.append(items['sex'])
        bmi.append(items['bmi'])
        children.append(items['children'])
        smoker.append(items['smoker'])
        region.append(items['region'])
        charges.append(items['charges'])

check if the data has been saved to the individual lists

In [7]:
smoker
print(len(smoker))

1338


We have 1,338 rows/observations in the dataset.

calculate the average of certain parameters from the dataset

create a function to calculate averages

In [8]:
def avg(lst):
    sum_items = 0
    for items in lst:
        sum_items += float(items)
    items_average = sum_items/len(lst)
    return items_average

### Age average

In [11]:
print(avg(age))

39.20702541106129


Average age of our dataset is 39 years

### Charges Average

In [12]:
print(avg(charges))

13270.422265141257


The average insurance charge paid is 13,270

Analyse which regions pay the most in Insurance Charges

Create a dictionary with region as keys and charge paid as values.
Create a separate lists for the charges incurred by the regions.
Find the sum of the different lists and compare.

In [13]:
zipped_region = zip(region, charges)
#print(list(zipped_region))
region_charges = {a:b for a,b in zipped_region}
print(region_charges)

{'southwest': '2007.945', 'southeast': '1629.8335', 'northwest': '29141.3603', 'northeast': '2205.9808'}


In [14]:
sum_female = []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['sex'] == 'female':
            sum_female.append(items['charges'])
print(sum_female)

['16884.924', '3756.6216', '8240.5896', '7281.5056', '28923.13692', '27808.7251', '11090.7178', '10797.3362', '13228.84695', '4149.736', '37701.8768', '14001.1338', '14451.83515', '12268.63225', '2198.18985', '4687.797', '15612.19335', '3046.062', '4949.7587', '6313.759', '3393.35635', '3556.9223', '12629.8967', '2211.13075', '3579.8287', '8059.6791', '13607.36875', '23244.7902', '5989.52365', '4133.64165', '14711.7438', '1743.214', '14235.072', '5920.1041', '16577.7795', '11741.726', '3947.4131', '2755.02095', '6571.02435', '7935.29115', '11033.6617', '39836.519', '43578.9394', '11073.176', '8026.6666', '11082.5772', '2026.9741', '10942.13205', '47291.055', '3766.8838', '12105.32', '6186.127', '21344.8467', '30942.1918', '5003.853', '2331.519', '11881.358', '2404.7338', '19107.7796', '8601.3293', '6686.4313', '2257.47525', '10115.00885', '3385.39915', '17081.08', '9634.538', '32734.1863', '12815.44495', '13616.3586', '11163.568', '2457.21115', '2155.6815', '27322.73386', '2166.732', '

In [15]:
sum_male = []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['sex'] == 'male':
            sum_male.append(items['charges'])
print(sum_male)

['1725.5523', '4449.462', '21984.47061', '3866.8552', '6406.4107', '2721.3208', '1826.843', '39611.7577', '1837.237', '2395.17155', '10602.385', '36837.467', '1137.011', '6203.90175', '2775.19215', '38711', '35585.576', '13770.0979', '51194.55914', '1625.43375', '2302.3', '39774.2763', '48173.361', '6272.4772', '6079.6715', '20630.28351', '38709.176', '23568.272', '37742.5757', '47496.49445', '34303.1672', '8606.2174', '4504.6624', '30166.61817', '6389.37785', '17663.1442', '6799.458', '11946.6259', '7726.854', '11356.6609', '1532.4697', '4441.21315', '37165.1638', '21098.55405', '30184.9367', '5729.0053', '10226.2842', '22412.6485', '15820.699', '3645.0894', '17560.37975', '3877.30425', '2867.1196', '47055.5321', '10825.2537', '4646.759', '11488.31695', '30259.99556', '11381.3254', '7740.337', '1705.6245', '39556.4945', '6082.405', '1632.56445', '1261.442', '2045.68525', '27375.90478', '3490.5491', '18972.495', '18157.876', '20745.9891', '40720.55105', '1842.519', '5125.2157', '7789.6

In [16]:
print(avg(sum_female))

12569.57884383534


In [17]:
print(avg(sum_male))

13956.751177721886


In [18]:
print(len(sum_female))
print(len(sum_male))

662
676


Male pay more in insurance charges compared to female. Both genders are both fairly represented.

In [19]:
smokers= []
sum_male = []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['smoker'] == 'yes':
            smokers.append(items['charges'])
print(smokers)

['16884.924', '27808.7251', '39611.7577', '36837.467', '37701.8768', '38711', '35585.576', '51194.55914', '39774.2763', '48173.361', '38709.176', '23568.272', '37742.5757', '47496.49445', '34303.1672', '23244.7902', '14711.7438', '17663.1442', '16577.7795', '37165.1638', '39836.519', '21098.55405', '43578.9394', '30184.9367', '47291.055', '22412.6485', '15820.699', '30942.1918', '17560.37975', '47055.5321', '19107.7796', '39556.4945', '17081.08', '32734.1863', '18972.495', '20745.9891', '40720.55105', '19964.7463', '21223.6758', '15518.18025', '36950.2567', '21348.706', '36149.4835', '48824.45', '43753.33705', '37133.8982', '20984.0936', '34779.615', '19515.5416', '19444.2658', '17352.6803', '38511.6283', '29523.1656', '12829.4551', '47305.305', '44260.7499', '41097.16175', '43921.1837', '33750.2918', '17085.2676', '24869.8368', '36219.40545', '46151.1245', '17179.522', '42856.838', '22331.5668', '48549.17835', '47896.79135', '42112.2356', '16297.846', '21978.6769', '38746.3551', '2487

In [20]:
non_smokers = []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['smoker'] == 'no':
            non_smokers.append(items['charges'])
print(non_smokers)

['1725.5523', '4449.462', '21984.47061', '3866.8552', '3756.6216', '8240.5896', '7281.5056', '6406.4107', '28923.13692', '2721.3208', '1826.843', '11090.7178', '1837.237', '10797.3362', '2395.17155', '10602.385', '13228.84695', '4149.736', '1137.011', '6203.90175', '14001.1338', '14451.83515', '12268.63225', '2775.19215', '2198.18985', '4687.797', '13770.0979', '1625.43375', '15612.19335', '2302.3', '3046.062', '4949.7587', '6272.4772', '6313.759', '6079.6715', '20630.28351', '3393.35635', '3556.9223', '12629.8967', '2211.13075', '3579.8287', '8059.6791', '13607.36875', '5989.52365', '8606.2174', '4504.6624', '30166.61817', '4133.64165', '1743.214', '14235.072', '6389.37785', '5920.1041', '6799.458', '11741.726', '11946.6259', '7726.854', '11356.6609', '3947.4131', '1532.4697', '2755.02095', '6571.02435', '4441.21315', '7935.29115', '11033.6617', '11073.176', '8026.6666', '11082.5772', '2026.9741', '10942.13205', '5729.0053', '3766.8838', '12105.32', '10226.2842', '6186.127', '3645.089

In [21]:
print(len(smokers))
print(len(non_smokers))

274
1064


We can safely say, based on the dataset that 25% are smokers while 75% are non smokers.

In [22]:
print(avg(smokers))
print(avg(non_smokers))

32050.23183153285
8434.268297856199


Smokers obviously pay more in insurance charges than snon smokers

#### Find the distinct regions in order to know which regions pay more insurance charges on average.

In [23]:
distinct_regions = []
for reg in region:
    if reg not in distinct_regions:
        distinct_regions.append(reg)
print(distinct_regions)

['southwest', 'southeast', 'northwest', 'northeast']


Four regions are represented in the dataset. Lets see which has the highest average charge.

In [24]:
southwest= []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['region'] == 'southwest':
            southwest.append(items['charges'])
print(southwest)

['16884.924', '1826.843', '1837.237', '10602.385', '36837.467', '4149.736', '38711', '35585.576', '4687.797', '51194.55914', '2302.3', '48173.361', '20630.28351', '23568.272', '1743.214', '14235.072', '11741.726', '7726.854', '39836.519', '11073.176', '47291.055', '12105.32', '15820.699', '6186.127', '5003.853', '2331.519', '11881.358', '4646.759', '7740.337', '17081.08', '9634.538', '6082.405', '11163.568', '1261.442', '2166.732', '1842.519', '10450.552', '5152.134', '4830.63', '8116.68', '5246.047', '48824.45', '10436.096', '8823.279', '5325.651', '6775.961', '4883.866', '5649.715', '7147.105', '13880.949', '1980.07', '8520.026', '7371.772', '5012.471', '5253.524', '34779.615', '1727.785', '12333.828', '35160.13457', '7196.867', '1832.094', '47305.305', '4260.744', '11837.16', '7441.053', '42856.838', '2150.469', '9625.92', '3172.018', '3906.127', '16297.846', '12646.207', '20177.67113', '8444.474', '1737.376', '34838.873', '35491.64', '2866.091', '47928.03', '9144.565', '13822.803',

In [25]:
southeast= []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['region'] == 'southeast':
            southeast.append(items['charges'])
            
print(southeast)

['1725.5523', '4449.462', '3756.6216', '8240.5896', '27808.7251', '11090.7178', '39611.7577', '1137.011', '14001.1338', '4949.7587', '6272.4772', '6313.759', '12629.8967', '38709.176', '37742.5757', '34303.1672', '23244.7902', '4504.6624', '5920.1041', '17663.1442', '16577.7795', '11946.6259', '3947.4131', '1532.4697', '37165.1638', '2026.9741', '3766.8838', '10226.2842', '30942.1918', '2867.1196', '47055.5321', '11381.3254', '19107.7796', '8601.3293', '2155.6815', '27322.73386', '18972.495', '5138.2567', '9877.6077', '7789.635', '21223.6758', '36950.2567', '19749.38338', '36149.4835', '13405.3903', '1631.8212', '7731.4271', '3981.9768', '12557.6053', '2137.6536', '1137.4697', '8516.829', '8871.1517', '37133.8982', '11743.299', '5002.7827', '2483.736', '3392.9768', '25081.76784', '19515.5416', '11987.1682', '2689.4954', '24227.33724', '14001.2867', '19444.2658', '1615.7667', '4463.2051', '7152.6714', '12648.7034', '1986.9334', '44260.7499', '5400.9805', '17085.2676', '20462.99766', '46

In [26]:
northwest= []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['region'] == 'northwest':
            northwest.append(items['charges'])
            
northwest

['21984.47061',
 '3866.8552',
 '7281.5056',
 '28923.13692',
 '6203.90175',
 '12268.63225',
 '2775.19215',
 '13770.0979',
 '1625.43375',
 '15612.19335',
 '3556.9223',
 '3579.8287',
 '8059.6791',
 '47496.49445',
 '5989.52365',
 '30166.61817',
 '4133.64165',
 '14711.7438',
 '6389.37785',
 '11356.6609',
 '6571.02435',
 '11033.6617',
 '21098.55405',
 '43578.9394',
 '8026.6666',
 '11082.5772',
 '10942.13205',
 '5729.0053',
 '17560.37975',
 '3877.30425',
 '10825.2537',
 '2404.7338',
 '6686.4313',
 '2257.47525',
 '10115.00885',
 '32734.1863',
 '1632.56445',
 '2045.68525',
 '18157.876',
 '20745.9891',
 '40720.55105',
 '10959.6947',
 '5125.2157',
 '6948.7008',
 '21348.706',
 '5028.1466',
 '6128.79745',
 '2719.27975',
 '2855.43755',
 '6455.86265',
 '11735.87905',
 '7419.4779',
 '4922.9159',
 '12044.342',
 '1639.5631',
 '9644.2525',
 '2130.6759',
 '13012.20865',
 '8162.71625',
 '3537.703',
 '10355.641',
 '7443.64305',
 '6710.1919',
 '17352.6803',
 '24476.47851',
 '43921.1837',
 '11520.09985',
 '33

In [27]:
northeast = []
with open('insurance.csv') as file:
    output= csv.DictReader(file)
    for items in output:
        if items['region'] == 'northeast':
            northeast.append(items['charges'])
            
northeast

['6406.4107',
 '2721.3208',
 '10797.3362',
 '2395.17155',
 '13228.84695',
 '37701.8768',
 '14451.83515',
 '2198.18985',
 '39774.2763',
 '3046.062',
 '6079.6715',
 '3393.35635',
 '2211.13075',
 '13607.36875',
 '8606.2174',
 '6799.458',
 '2755.02095',
 '4441.21315',
 '7935.29115',
 '30184.9367',
 '22412.6485',
 '3645.0894',
 '21344.8467',
 '11488.31695',
 '30259.99556',
 '1705.6245',
 '39556.4945',
 '3385.39915',
 '12815.44495',
 '13616.3586',
 '2457.21115',
 '27375.90478',
 '3490.5491',
 '6334.34355',
 '19964.7463',
 '7077.1894',
 '15518.18025',
 '10407.08585',
 '4827.90495',
 '1694.7964',
 '8538.28845',
 '4005.4225',
 '43753.33705',
 '14901.5167',
 '4337.7352',
 '20984.0936',
 '6610.1097',
 '10564.8845',
 '7358.17565',
 '9225.2564',
 '38511.6283',
 '5354.07465',
 '29523.1656',
 '4040.55825',
 '12829.4551',
 '41097.16175',
 '13047.33235',
 '24869.8368',
 '14590.63205',
 '9282.4806',
 '9617.66245',
 '9715.841',
 '22331.5668',
 '48549.17835',
 '4237.12655',
 '11879.10405',
 '9432.9253',
 

In [28]:
print(avg(southwest))
print(avg(southeast))
print(avg(northwest))
print(avg(northeast))

12346.93737729231
14735.411437609895
12417.575373969228
13406.3845163858


In [29]:
print(len(southwest))
print(len(southeast))
print(len(northwest))
print(len(northeast))

325
364
325
324


The southeast have the highest average insurance charges and also the highest number of people paying insurance charges.

We can also notice that all the regions are fairly represented with shows good standards of data collection.

#### Create a dictionary for our averages

In [31]:
lst_1 = ['avg_age', 'avg_bmi', 'avg_children', 'avg_charges']
lst_2 = [avg(age), avg(bmi), avg(children), avg(charges)]
averages = {a:b for a,b in zip(lst_1, lst_2)}
averages

{'avg_age': 39.20702541106129,
 'avg_bmi': 30.663396860986538,
 'avg_children': 1.0949177877429,
 'avg_charges': 13270.422265141257}

We can see the averages of all the numerical data in the averages dictionary.